bitbots_hcm/src/bitbots_hcm/training/train.py

```diff
@@ -0,0 +1,219 @@
+    classifier = FallChecker(thresh_gyro_front=thresh_gyro_front, thresh_gyro_side=thresh_gyro_side,
+                             thresh_orient_front=thresh_orient_front, tresh_orient_side=tresh_orient_side)
+    return evaluate_classifier(classifier)
+
+
+def pickle_classifier(classifier, type):
+    path = "classifiers/" + type + "/"
+    Path(path).mkdir(parents=True, exist_ok=True)
+    with open(path + "classifier.pkl", "wb") as file:
+        pickle.dump(classifier, file)
+    with open(path + "scaler.pkl", "wb") as file:
+        pickle.dump(scaler, file)
+    with open(path + "types.pkl", "wb") as file:
+        pickle.dump(data_types, file)
+
+
+if args.knn:
+    objective = knn
+    name = 'knn'
+elif args.mlp:
+    objective = mlp
+    name = 'mlp'
+elif args.svc:
+    objective = svc
+    name = 'svc'
+elif args.gpc:
+    objective = gpc
+    name = 'gpc'
+elif args.bb:
+    objective = bb
+    name = 'bb'
+
+seed = np.random.randint(2 ** 32 - 1)
+sampler = TPESampler(n_startup_trials=10, seed=seed)
+if use_fn:
+    direction = 'minimize'
+else:
+    direction = 'maximize'
+study = optuna.create_study(study_name=args.name, storage=args.storage, direction=direction,
+                            sampler=sampler, load_if_exists=True)
+print("Start optimizing hyper parameters")
+study.optimize(objective, n_trials=args.iterations, show_progress_bar=True, n_jobs=1)
+
+print("Train final classifier with optimized hyper parameters")
+if args.knn:
+    classifier = sklearn.neighbors.KNeighborsClassifier(n_neighbors=study.best_params['n_neighbors'],
+                                                        weights=study.best_params['weights'],
+                                                        algorithm=study.best_params['algorithm'],
+                                                        leaf_size=study.best_params['leaf_size'])
+elif args.mlp:
+    hidden_layer = (study.best_params['hidden_layer_size'],) * study.best_params['hidden_layer_count']
+    classifier = MLPClassifier(hidden_layer_sizes=hidden_layer, activation=study.best_params['activation_fun'],
+                               alpha=study.best_params['alpha'], learning_rate=study.best_params['learning_rate'])
+elif args.svc:
+    classifier = sklearn.svm.SVC(C=study.best_params['svc_c'], gamma="auto", kernel=study.best_params['kernel'],
+                                 degree=study.best_params['degree'])
+elif args.gpc:
+    classifier = GaussianProcessClassifier(multi_class=study.best_params['multi_class'])
+elif args.bb:
+    classifier = FallChecker(thresh_gyro_front=study.best_params['thresh_gyro_front'],
+                             thresh_gyro_side=study.best_params['thresh_gyro_side'],
+                             thresh_orient_front=study.best_params['thresh_orient_front'],
+                             tresh_orient_side=study.best_params['tresh_orient_side'])
+classifier.fit(x_scaled, y)
```

Comment:
In the case of the bb classifier this would fail, right? Because the `FallChecker` class has no methods such as `.fit()` or `.predict()`. Or is there something I overlooked?

Reply:
Good catch! I actually forgot to commit one file, should be fixed now.